spec/system/distribution_system_spec.rb

```diff
@@ -0,0 +1,235 @@
+          end
+          page.find ".alert"
+        end.to change { Distribution.count }.by(-1).and change { Item.count }.by(1)
+        expect(page).to have_content "reclaimed"
+      end
+    end
+  end
+
+  context "When creating a distribution and items have value" do
+    before do
+      item1 = create(:item, value: 10.5)
+      item2 = create(:item)
+      item3 = create(:item, value: 1)
+      @distribution1 = create(:distribution, :with_items, item: item1, agency_rep: "A Person", organization: @user.organization)
+      create(:distribution, :with_items, item: item2, agency_rep: "A Person", organization: @user.organization)
+      @distribution3 = create(:distribution, :with_items, item: item3, agency_rep: "A Person", organization: @user.organization)
+      visit @url_prefix + "/distributions"
+    end
+
+    it 'the user sees value in row on index page' do
+      # row: 100 items * 1$
+      expect(page).to have_content "$100"
+    end
+
+    it 'the user sees total value on index page' do
+      # 100 items * 10.5 + 100 items * 1
+      expect(page).to have_content "$1,150"
+    end
+
+    it 'the user sees value per item on show page' do
+      # item value 10.50
+      visit @url_prefix + "/distributions/#{@distribution1.id}"
+      expect(page).to have_content "$10.50"
+    end
+
+    it 'the user sees total value on show page' do
+      # 100 items * 10.5
+      visit @url_prefix + "/distributions/#{@distribution1.id}"
+      expect(page).to have_content "$1,050"
+    end
+  end
+
+  context "When creating a distribution from a donation" do
+    let(:donation) { create :donation, :with_items }
+    before do
+      visit @url_prefix + "/donations/#{donation.id}"
+      click_on "Start a new Distribution"
+      within "#new_distribution" do
+        select @partner.name, from: "Partner"
+        click_button "Save"
+      end
+    end
+
+    it "it completes successfully" do
+      expect(page).to have_content "Distributions"
+      expect(page.find(".alert-info")).to have_content "reated"
+      expect(Distribution.first.line_items.count).to eq 1
+    end
+
+    context "when editing that distribution" do
+      before do
+        click_on "Edit", match: :first
+        @distribution = Distribution.last
+      end
+
+      it "User creates a distribution from a donation then edits it" do
+        within "#edit_distribution_#{@distribution.to_param}" do
+          first(".numeric").set 13
+          click_on "Save"
+        end
+        expect(page).to have_content "Distribution updated!"
+        expect(page).to have_content 13
+      end
+
+      it "User creates a distribution from a donation then tries to make the quantity too big" do
+        within "#edit_distribution_#{@distribution.to_param}" do
+          first(".numeric").set 999_999
+          click_on "Save"
+        end
+        expect(page).to have_no_content "Distribution updated!"
+        expect(page).to have_content "Distribution could not be updated!"
```

Comment:
This is going to fail because the error text has changed
```suggestion
        expect(page).to have_content "Insufficient Supply"
```